bundles/org.eclipse.swt/Eclipse SWT Tests/win32/org/eclipse/swt/widgets/DisplayWin32Test.java

```diff
@@ -0,0 +1,55 @@
+package org.eclipse.swt.widgets;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.eclipse.swt.internal.*;
+import org.eclipse.swt.internal.win32.*;
+import org.junit.jupiter.api.*;
+
+public class DisplayWin32Test {
+
+	private Display display;
+
+	@BeforeAll
+	public static void assumeIsFittingPlatform() {
+		PlatformSpecificExecution.assumeIsFittingPlatform();
+	}
+
+	@BeforeEach
+	public void createDisplay() {
+		display = new Display();
+	}
+
+	@AfterEach
+	public void destroyDisplay() {
+		display.dispose();
+	}
+
+	@Test
+	public void setRescaleAtRuntime_activate() {
+		display.setRescalingAtRuntime(true);
+		assertTrue(display.isRescalingAtRuntime());
+		assertEquals(OS.DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2, OS.GetThreadDpiAwarenessContext());
+	}
+
+	@Test
+	public void setRescaleAtRuntime_deactivate() {
+		display.setRescalingAtRuntime(false);
+		assertFalse(display.isRescalingAtRuntime());
+		assertEquals(OS.DPI_AWARENESS_CONTEXT_SYSTEM_AWARE, OS.GetThreadDpiAwarenessContext());
+	}
+
+	@Test
+	public void setRescaleAtRuntime_toggling() {
+		display.setRescalingAtRuntime(false);
+		assertFalse(display.isRescalingAtRuntime());
+		assertEquals(OS.DPI_AWARENESS_CONTEXT_SYSTEM_AWARE, OS.GetThreadDpiAwarenessContext());
```

Comment:
There is no default value. With rescaling at runtime deactivated, the DPI awareness is set to "system" (as that's the mode for which the existing HiDPI support was developed). For that reason, the OS class currently already makes the process DPI aware at startup.
Currently, it would be possible to start the application with a custom DPI awareness (which already has prefence over the old "DPI aware" property) and then calling `setRescaleAtRuntime(false)` for the main display would change that custom DPI awareness to "system", but I do not see why anyone should want the application to run in an unsupported DPI awareness mode at all.

Reply:
I see you point but this test is testing 2 things right now:
1. Changing the mode
2. What's the default mode

And since there is no default mode (just a de-facto default that is given by the JVM / OS) then it's not correct to test for it and enforce it in a test. There's nothing to win and something to loose: the test will unnecessarily break if the de-facto default changes in the future.

If testing that the (de-facto) default is "System" is important then it should be done in another test method.

Reply:
Please read my previous comment. The test does not rely on a default (of the JVM/OS/whatever), it explicitly sets the values to be tested from the beginning.

Reply:
Ah, I see what you mean: by calling `setRescaleAtRuntime(false)` you're already setting the DPI awareness to `OS.DPI_AWARENESS_CONTEXT_SYSTEM_AWARE` ("system").

All good, thank you for the clarification.